fog/report/connection/src/lib.rs

```diff
@@ -1,172 +1,104 @@
 // Copyright (c) 2018-2020 MobileCoin Inc.
 
-//! Fog Report Connection handles resolving the public key from the report service
-//! on making grpc to the reports endpoint
+#![deny(missing_docs)]
+
+//! Fog Report Connection handles connecting to the fog report service and
+//! building up a FogReportResponses object needed to create a transaction
+//! with fog recipients.
 
-use core::str::FromStr;
 use displaydoc::Display;
 use grpcio::{ChannelBuilder, Environment};
-use mc_account_keys::PublicAddress;
-use mc_attest_core::{VerificationReport, Verifier};
 use mc_common::logger::{log, o, Logger};
 use mc_fog_api::report_grpc;
-use mc_fog_report_validation::ingest_report::{Error as IngestReportError, IngestReportVerifier};
-use mc_util_grpc::{BasicCredentials, ConnectionUriGrpcioChannel};
-use mc_util_uri::{ConnectionUri, FogUri, UriParseError};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::FogUri;
 use std::sync::Arc;
 
-pub use mc_fog_report_validation::{FogPubkeyResolver, FullyValidatedFogPubkey};
+pub use mc_fog_report_validation::FogReportResponses;
+
+/// Fog report server connection based on grpcio
+///
+/// TODO: As an optimization, it might be good to make this like a connection
+/// pool that holds onto its grpc connections, so that they can be reused
+/// if a client is constructing transactions frequently.
+/// The user of this API would tear down this object when they want these
+/// connections to be closed.
```

Comment:
but the implementation here creates a new grpc channel every time we fetch -- i think for grpc to provide connection reuse, we have to keep the channel object alive across calls

Reply:
https://tikv.github.io/doc/grpcio/struct.Channel.html

> Channels are an abstraction of long-lived connections to remote servers. More client objects can reuse the same channel.

So like, ideally, if mobilecoind is sending to many fog users in succession, it would re-use the tcp connection each time it hits `fog.mobilecoin.com` or whatever. I think we have to keep the `Channel` object alive for that to happen, I think the grpc connection pool is in the channel object

Reply:
i can rewrite the code comment